src/shared/ui/inputPrompter.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { applyPrimitives } from '../utilities/collectionUtils'
+import { WIZARD_BACK } from '../wizards/wizard'
+import { QuickInputButton, PrompterButtons } from './buttons'
+import { Prompter, PromptResult } from './prompter'
+
+/** Additional options to configure the `InputBox` beyond the standard API */
+interface AdditionalInputBoxOptions {
+    title?: string
+    step?: number
+    placeholder?: string
+    totalSteps?: number
+    buttons?: PrompterButtons<string>
```

Comment:
The default to `WizardControl` was supposed to be for ergonomics since it will always be able to accept a Back button, though I might revisit this. For implementations you will almost never need to explicitly declare what kind of buttons you're using since it can be inferred quite readily by the compiler.